crates/ruff_linter/src/rules/pygrep_hooks/rules/blanket_noqa.rs

```diff
@@ -88,8 +88,13 @@ pub(crate) fn blanket_noqa(
 ) {
     for directive_line in noqa_directives.lines() {
         if let Directive::All(all) = &directive_line.directive {
-            let line = locator.slice(directive_line.range);
-            let offset = directive_line.range.start();
+            let line = if directive_line.end() > locator.contents().text_len() {
```

Comment:
This seems like a bug, if the `.end()` is greater than the contents... Have you looked into how that happens?

Reply:
I see the PR summary, but what happens if we remove it?

Reply:
Ya it generates a fixture failure with W292_1.py

Reply:
It seems the code mentioned in the PR summary was added by @MichaReiser so maybe he can shed some light.